test/unit/JobsManager.js

```diff
@@ -658,205 +720,281 @@ contract("JobsManager", accounts => {
 
-            await expectThrow(jobsManager.missedVerificationSlash(jobId, claimId, segmentNumber, {from: accounts[2]}))
+            await expectThrow(jobsManager.missedVerificationSlash(0, 0, 0, {from: watcher}))
         })
 
-        it("should update job escrow, refund broadcaster deposit and set claim as slashed", async () => {
-            // Fast foward through verification period
-            const verificationPeriod = await jobsManager.verificationPeriod.call()
-            await fixture.roundsManager.mineBlocks(verificationPeriod.toNumber() + 1)
+        it("should refund the broadcaster", async () => {
+            // Fast forward through verification period
+            const endVerificationBlock = (await jobsManager.getClaim(0, 0))[3]
+            await fixture.roundsManager.setMockUint256(functionSig("blockNum()"), endVerificationBlock.add(1))
 
-            await jobsManager.missedVerificationSlash(jobId, claimId, segmentNumber, {from: accounts[2]})
+            await jobsManager.missedVerificationSlash(0, 0, 0, {from: watcher})
 
-            const jInfo = await jobsManager.getJob(jobId)
-            const jEscrow = jInfo[8]
-            assert.equal(jEscrow, 0, "escrow is incorrect")
             const bDeposit = (await jobsManager.broadcasters.call(broadcaster))[0]
-            assert.equal(bDeposit, 1000, "broadcaster deposit is incorrect")
-            const cInfo = await jobsManager.getClaim(jobId, claimId)
-            const cStatus = cInfo[5]
-            assert.equal(cStatus, 1, "claim status is incorrect")
+            assert.equal(bDeposit, 1000, "wrong broadcaster deposit")
+            const jEscrow = (await jobsManager.getJob(0))[8]
+            assert.equal(jEscrow, 0, "wrong job escrow")
         })
 
-        it("should throw if claim is not pending", async () => {
-            // Fast foward through verification period
-            const verificationPeriod = await jobsManager.verificationPeriod.call()
-            await fixture.roundsManager.mineBlocks(verificationPeriod.toNumber() + 1)
+        it("should set the claim as slashed", async () => {
+            // Fast forward through verification period
+            const endVerificationBlock = (await jobsManager.getClaim(0, 0))[3]
+            await fixture.roundsManager.setMockUint256(functionSig("blockNum()"), endVerificationBlock.add(1))
 
-            await jobsManager.missedVerificationSlash(jobId, claimId, segmentNumber, {from: accounts[2]})
+            await jobsManager.missedVerificationSlash(0, 0, 0, {from: watcher})
 
-            // Should fail because claim is already slashed
-            await expectThrow(jobsManager.missedVerificationSlash(jobId, claimId, segmentNumber, {from: accounts[2]}))
+            const cStatus = (await jobsManager.getClaim(0, 0))[5]
+            assert.equal(cStatus, 1, "wrong claim status")
         })
     })
 
     describe("doubleClaimSegmentSlash", () => {
```

Comment:
We should probably make sure the transcoder can be slashed in any status (add it to bondingManager.js?)

Reply:
Yep planning on adding the tests for the BondingManager unit test